pytorch_lightning/accelerators/gpu.py

```diff
@@ -33,6 +33,18 @@
 class GPUAccelerator(Accelerator):
     """Accelerator for GPU devices."""
 
+    def __init__(self) -> None:
+        """
+        Raises:
+            MisconfigurationException:
+                If torch.cuda isn't available.
+                If no CUDA devices are found.
+        """
+        if not torch.cuda.is_available():
+            raise MisconfigurationException("GPU Accelerator used, but CUDA isn't available.")
+        if torch.cuda.device_count() == 0:
+            raise MisconfigurationException("GPU Accelerator used, but found no CUDA devices available.")
```

Comment:
@carmocca @awaelchli going through #11448, this logic is still replicated across the trainer & each accelerator to support `accelerator="auto"`

What do you think of this instead?
```py
Accelerator(ABC):

@staticmethod
@abstractmethod
def is_available() -> bool:

def setup_environment(self):
    if not self.is_available():
        raise MisconfigurationException("{self.__class__.__qualname__} is not configured to run in this environment")
```

This would function pretty much the same as how `ClusterEnvironment.detect()` works, which means we don't need to duplicate logic across individual accelerators & the accelerator connector.

And the runtime check during trainer execution is automatically applied without being a breaking change that this PR currently introduces.

cc @four4fish